ztype/array_encode_test.go

```diff
@@ -117,16 +113,13 @@ func TestBitfieldArrayEncoding(t *testing.T) {
 				IsAuto:      test.isAuto,
 				IsPacked:    test.isPacked,
 			}
-			// convert to binary buffer
-			var buf bytes.Buffer
-			w := bitio.NewCountWriter(&buf)
-			err := arrayInstance.MarshalZserio(w)
-			w.Close()
+
+			got, err := zserio.Marshal(&arrayInstance)
 			if err != nil {
 				t.Fatalf("error writing array: %v", err)
 			}
 
-			if diff := cmp.Diff(test.want, buf.Bytes()); diff != "" {
+			if diff := cmp.Diff(test.want, got); diff != "" {
 				t.Errorf("incorrect encoding: %s", diff)
 			}
 		})
```

Comment:
There seems to be a lot of repetition here. Can we factor a function like `func verifyMarshal(t *testing.T, want []byte, obj zserio.Marshal)` ?

Reply:
Done.